src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...currentState };
```

Comment:
The name 'newState' for the object copy is not descriptive enough. Consider renaming it to 'stateCopy' or something similar that indicates it is a copy of the current state.